internal/health/core.go

```diff
@@ -43,8 +47,8 @@ func (c *Core) IsHealthy() bool {
 			continue
 		}
 		isHealthy, err := checker.checkHealth()
-		if !isHealthy {
-			logger.Ctx(context.TODO()).Errorw("health check failed", "msg", err)
+		if !isHealthy || err != nil {
```

Comment:
break the err case handling in separate if block
```suggestion
		if err != nil {
		log err.Error
        handle err based on err types and decide to return true/false
		}
		
		return isHealthy
```

Reply:
If you look at the full code snippet these checks are running inside a loop. So having `!isHealthy || err != nil` is a fail-fast implementation. In the case of healthy checks, we just let the loop complete and return from the end of the function.